File: chat-1/cpp/project/src/chat-1.cpp
#include <string>

#include <boost/shared_ptr.hpp>

#include <rsb/Event.h>
#include <rsb/Factory.h>
#include <rsb/Handler.h>
#include <rsb/filter/OriginFilter.h>

using namespace std;

using namespace boost;

void printMessage(rsb::EventPtr event) {
    // Insert display of messages
}

int main(int argc, char *argv[]) {
    if (argc != 2) {
	cerr << "usage: " << argv[0] << " NICKNAME" << endl;
	return EXIT_FAILURE;
    }
    string nick = argv[1];
    // Insert creation of sending and receiving participants
    rsb::Factory &factory = ...

    rsb::Informer<string>::Ptr informer = ...
    rsb::ListenerPtr listener = ...
    listener->addFilter(rsb::filter::FilterPtr(new rsb::filter::OriginFilter(informer->getId(), true)));
    // Add a handler
    listener->addHandler(...);

    while (true) {
	cout << "> ";
	cout.flush();
	shared_ptr<string> message(new string());
	getline(cin, *message);
	if (*message == "/quit") {
	    break;
	}
	// Send the message
	informer ...
    }

    return EXIT_SUCCESS;
}

File: chat-1/python/project/chat1.py
import sys
import rsb
import rsb.filter

def chatClient(nick):
    # Insert creation of sending and receiving participants
    informer = ...
    listener = ...

    def printMessage(event):
	# Insert display of messages
	...

    listener.addFilter(rsb.filter.OriginFilter(informer.id, invert = True))
    # Add a handler
    listener.addHandler(...)

    while True:
        sys.stdout.write('> ')
        sys.stdout.flush()
        line = sys.stdin.readline().strip()
        if line == '/quit':
            return
        if line:
            # Insert sending code here
            informer ...

if __name__ == '__main__':
    if len(sys.argv) < 2:
        print 'usage: %s NICKNAME' % sys.argv[0]
        sys.exit(1)
    nick = sys.argv[1]

    chatClient(nick)
